pre-commit.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# No tests to update gh-pages
+BRANCH=$(git symbolic-ref HEAD 2>/dev/null)
+[ $BRANCH = "refs/heads/gh-pages" ] && exit 0
+
+git stash save -k -q
+make
```

Comment:
I think that it's only necessary to build one version, though it probably won't save much time overall.  Maybe a `make check` target would save some time.